src/payment/strategies/digitalriver/digitalriver-payment-strategy.ts

```diff
@@ -0,0 +1,167 @@
+
+        return state;
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    async execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = orderRequest;
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+
+        if (!payment || !this._loadSuccessResponse || !this._digitalRiverCheckoutId) {
+            throw new InvalidArgumentError('Unable to proceed because payload payment argument is not provided.');
+        }
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    private _getDigitalRiverJs(): DigitalRiverJS {
+        if (!this._digitalRiverJS) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return this._digitalRiverJS;
+    }
+
+    private _getErrorMessage(error: OnCancelOrErrorResponse): string {
+        const { errors } = error;
+
+        return errors.map(e => 'code: ' + e.code + ' message: ' + e.message).join('\n');
+    }
+
+    private _onSuccessResponse(data?: OnSuccessResponse): Promise<void> {
+        return new Promise((resolve, reject) => {
+            if (data && this._submitFormEvent) {
+                const { browserInfo } = data.source;
+                this._loadSuccessResponse = browserInfo ? {
+                    source: {
+                        id: data.source.id,
+                        reusable: data.source.reusable,
+                        ...browserInfo,
+                    },
+                    readyForStorage: data.readyForStorage,
+                } : {
+                    source: {
+                        id: data.source.id,
+                        reusable: data.source.reusable,
+                    },
+                    readyForStorage: data.readyForStorage,
+                } ;
+                resolve(this._submitFormEvent());
+            }
+
+            reject(this._getDigitalRiverInitializeOptions().onError?.
+                (new InvalidArgumentError('Unable to initialize payment because success argument is not provided.')));
+        });
+    }
+
+    private _onReadyResponse(data?: OnReadyResponse): void {
+        if (data) {
+            this._getDigitalRiverInitializeOptions().onRenderButton?.();
+        }
+    }
+
+    private _getDigitalRiverInitializeOptions(): DigitalRiverPaymentInitializeOptions {
+        if (!this._digitalRiverInitializeOptions) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return this._digitalRiverInitializeOptions;
+    }
+
+    private _getCustomerOptions(customer?: Customer): Customer {
+        if (!customer) {
```

Comment:
rather than checking this in the function, why don't you instead assume that `customer` is non-nullable

Reply:
Hey @capsula4 , yes I could remove that but I'll need a new validation for this email:  

> const configuration = {
>             sessionId: clientToken.sessionId,
>             options: { ...this._getDigitalRiverInitializeOptions().configuration },
>             billingAddress: {
>                 firstName: billing.firstName,
>                 lastName: billing.lastName,
>                 // tslint:disable-next-line:no-non-null-assertion
>                 email: billing.email || customer!.email,
>                 phoneNumber: billing.phone,
.........

what do you think about it? or Do you have another way in mind?

```
                // tslint:disable-next-line:no-non-null-assertion
                email: billing.email || customer!.email,
```

Reply:
Thanks @capsula4, I think It's ready for another review ♻️